src/Agent.Worker/Variables.cs

```diff
@@ -589,6 +637,58 @@ public void CopyInto(Dictionary<string, VariableValue> target)
             }
         }
 
+        private Boolean IsReadOnly(Variable variable)
+        {
+            if (variable.ReadOnly)
+            {
+                return true;
+            }
+
+            if (_wellKnownSystemVariables == null)
```

Comment:
This is the weirdest part of the PR, it basically grabs all of the fields from a bunch of different classes and tosses them into this list. If there's a better way to do this, let me know

Reply:
Is the reason we don't just look for the prefix the fact that i can name my own var `system.myvar`?

Reply:
Yep

Reply:
I admire what you're doing here, cause it's fancy :-),

I'm wondering if just declaring a List of ReadOnlyVariables in Constants.cs makes sense.  Things would be more explicit.

Reply:
I thought about it - the big advantage here is that we get to pull in `WellKnownDistributedTaskVariables` which comes from the server, so we can pick up server side changes whenever the contract changes without having to update locally. So this ends up being the set of all variables on both the agent and the server

Reply:
Generally we try to avoid reflection if we can, since it's slow and skips the complier.

If someone added new members to WellKnownDistributedTaskVariables on the server side would they expect them to be iterated in this fashion?  Are there any new vars or types someone could add to WellKownDistributedTaskVariables that would cause an exception in the code below?

While having a declared list of ReadOnlyVariables is verbose, it's also more explicit and avoids reflection fun.